### PR TITLE
Adding GMP and Cloud Monitoring for gke-batch-refarch

### PR DESCRIPTION
**Pull request by JamesDuncanNz**
Adding support for GMP Managed Collection and Cloud Monitoring Dashboard for gke-batch-refarch.

### CODE REVIEW

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Monitoring Kueue with Google Managed Prometheus and Cloud Monitoring
+
+This document describes how to monitor Kueue metrics using Google Managed Prometheus and Cloud Monitoring.
+
+## Overview
+
+Kueue metrics are automatically collected by Google Managed Prometheus using managed collection. This means Prometheus-based collectors run as a Daemonset, ensuring scalability by only scraping targets on colocated nodes. The collected metrics are then exported and made available in Google Cloud's Monitoring service.
```
Comment (alculquicondor) [inline best-practices/gke-batch-refarch/02_platform/monitoring/gmp/README.md:7]:
```suggestion
You can configure Google Managed Prometheus to automatically collect Kueue metrics. The collected metrics are then exported and made available in Google Cloud's Monitoring service.
```

The rest seems to be documentation for GMP itself, which is not relevant here.

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+## Viewing the Dashboard
+
+The Kueue dashboard is available in Google Cloud Monitoring. This dashboard provides a visual representation of key Kueue metrics, allowing you to quickly assess the health and performance of your Kueue system.
+
+<img src="../../../images/kueue_cloud_monitoring_1.png" width="800">
+<img src="../../../images/kueue_cloud_monitoring_2.png" width="800">
+
+## Configuring Managed Collection and Creating the Dashboard
+
+The following command will configure Managed Collection for Kueue and Create the Dashboard in Cloud Monitoring.
```
Comment (alculquicondor) [inline best-practices/gke-batch-refarch/02_platform/monitoring/gmp/README.md:18]:
```suggestion
Run the following command to configure Managed Collection for Kueue and Create the Dashboard in Cloud Monitoring. 
```

Also, shouldn't this section be before "Viewing the Dashaboard"?

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/kueue-dashboard.json
Comment (alculquicondor) [inline best-practices/gke-batch-refarch/02_platform/monitoring/gmp/kueue-dashboard.json:1]: is this the same dashboard as the one outside of the gmp folder?

Should we remove the other mechanism? The Ref architecture is supposed to be opinionated.

Comment (JamesDuncanNz) [reply best-practices/gke-batch-refarch/02_platform/monitoring/gmp/kueue-dashboard.json:1]: The dashboard is slightly different, but was heavily inspired from the dashboard in the parent folder. I had to use some additional metrics that weren't available in Cloud Monitoring and added a graph or two.

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/gmp-kueue-monitoring.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  name: gmp-system:collector:kueue-secret-reader
+  namespace: kueue-system
+roleRef:
+  name: kueue-secret-reader
+  kind: Role
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- name: collector
+  namespace: gmp-system
+  kind: ServiceAccount
```
Comment (laurentgrangeau) [inline best-practices/gke-batch-refarch/02_platform/monitoring/gmp/gmp-kueue-monitoring.yaml:77]: I suggest adding a line break

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/install-gmp.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[[ ! "${PROJECT_ID}" ]] && echo -e "Please export PROJECT_ID variable (\e[95mexport PROJECT_ID=<YOUR POROJECT ID>\e[0m)\nExiting." && exit 0
```
Comment (laurentgrangeau) [inline best-practices/gke-batch-refarch/02_platform/monitoring/gmp/install-gmp.sh:1]: I suggest adding a copyright header for this code

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/install-gmp.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+echo -e "\e[95mPROJECT_ID is set to ${PROJECT_ID}\e[0m"
+
+[[ ! "${REGION}" ]] && echo -e "Please export REGION variable (\e[95mexport REGION=<YOUR REGION, eg: us-central1>\e[0m)\nExiting." && exit 0
+echo -e "\e[95mREGION is set to ${REGION}\e[0m"
+
+kubectl apply -f gmp-kueue-monitoring.yaml && \
+gcloud monitoring dashboards create --project=$PROJECT_ID --config-from-file=kueue-dashboard.json
```
Comment (laurentgrangeau) [inline best-practices/gke-batch-refarch/02_platform/monitoring/gmp/install-gmp.sh:8]: I suggest adding a line break

**Review by laurentgrangeau — CHANGES_REQUESTED** [review]
Hey, some minor changes and it should be good to go

**Review by alculquicondor — APPROVED** [review]

File: best-practices/gke-batch-refarch/02_platform/monitoring/gmp/kueue-dashboard.json
Comment (JamesDuncanNz) [reply best-practices/gke-batch-refarch/02_platform/monitoring/gmp/kueue-dashboard.json:1]: @laurentgrangeau Are you ok with this?

Comment (laurentgrangeau) [reply best-practices/gke-batch-refarch/02_platform/monitoring/gmp/kueue-dashboard.json:1]: Yes, I'm OK with this

**Pull request merged**